### PR TITLE
Add some documentation to Memoro

### PR DESCRIPTION
**Pull request by rrtheonlyone**
This PR helps to add two more files to Memoro which cover detailed instructions on set-up. This will hopefully help ease the difficulty in set-up process for a new developer.

There are new changes to memoro and the docs will have to be updated to reflect that (e.g. sampling). However, this is def a start. 

I can make any extra changes if necessary.

### Changelog:
- Added docs/set_up.md for setting up
- Added docs/use_case.md for a use case with memoro
- Updated README.md
- Updated gitignore
- Added new asset (image) for documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,7 +34,7 @@ For the main LLVM repo, the Clang repo, and the CompilerRT repo, use the Memoro
 [CompilerRT](https://github.com/epfl-vlsc/compiler-rt)
 
 These repos should default to branch r40\_dev (Memoro is based off of LLVM release\_40).
-Optionally compile with libcxx and libcxxabi. 
+Optionally compile with libcxx and libcxxabi. If this does not work, use branch memoro_80.
```
Comment (ttreyer) [inline README.md]: The branch memoro_80 is the active branch. We should direct people directly to memoro_80.

**Comment by ttreyer** [comment]
Nice job!
It would be neat to mentions Memoro's environment variables in `docs/use_case.md`, as the default setting is to not collect any data.
We have two important variables: `MEMORO_SYMBOLIZER_PATH`, and `MEMORO_OPTIONS` which takes multiple flags as described in `compiler-rt/lib/memoro/memoro_flags.inc`.
A typical configuration would be:
``` bash
MEMORO_SYMBOLIZER_PATH=path-to-llvm/build/bin/llvm-symbolizer \
MEMORO_OPTIONS=register_allocs=1,access_sampling_rate=1 \
./my-binary …
```

File: README.md (hunk 1 above)
Comment (sbyma) [reply README.md]: I second this, r40_dev is pretty obsolete at this point.

**Review by sbyma — APPROVED** [review]
I think it all looks fine. We should just remove any references to r40_dev. The VLSC llvm/clang repos should already default to the memoro_80 branch.

File: README.md (hunk 1 above)
Comment (rrtheonlyone) [reply README.md]: No worries, have updated accordingly!

**Comment by rrtheonlyone** [comment]
@ttreyer No problem, I will update `docs/use_case.md` to include those flags as well!

**Pull request merged**